src/Tizen.NUI.WindowSystem/src/public/SoftkeyClient.cs

```diff
@@ -61,7 +62,44 @@ public SoftkeyClient(TizenShell tzShell, Window win)
 
             _tzsh = tzShell;
             _tzshWin = win.GetNativeId();
-            _softkeyClient = Interop.SoftkeyClient.Create(_tzsh.GetNativeHandle(), (IntPtr)_tzshWin);
+            _softkeyClient = Interop.SoftkeyClient.Create(_tzsh.GetNativeHandle(), (uint)_tzshWin);
+            if (_softkeyClient == IntPtr.Zero)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _tzsh.ErrorCodeThrow(err);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new Softkey Client handle.
+        /// </summary>
+        /// <param name="tzShell">The TizenShell instance.</param>
+        /// <param name="win">The window provider for the quickpanel service.</param>
+        /// <privilege>http://tizen.org/privilege/windowsystem.admin</privilege>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when a argument is null.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation or no service.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the caller does not have privilege to use this method.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <since_tizen> 12 </since_tizen>        
+        public SoftkeyClient(TizenShell tzShell, IWindowProvider win)
```

Comment:
seeing how tzShell and IWindowProvider are passed together I wonder if TizenShell could provide IWindowProvider to the clients/services, or would that be improper coupling?

Reply:
@wiertel 
Hello, I have just copied the previous interface of method. 
https://github.com/Samsung/TizenFX/pull/6182/files#diff-012369306f465077399c17ed4738f4c3b471643f87404ce97d957c5a4ca0331bL47

@juns-kim 
Hello, can you check and answer the Mr. @wiertel 's question?
thank you.

Reply:
Seems like it increases the coupling between TizenShell and Window.
The TizenShell simply handles the Tizen-ws-shell protocol and doesn't contain any window information.

Reply:
SoftkeyClient는 이미 Ecore_Wl2_Window에 의존성이 있는 것이 아닌가요?

window id자체가 Wl2에만 특정한 값인데, 이 값을 IWindow에 추가하기 보다는 이 값을 얻어 올 수 있는 Ecore_Wl2_Window를 통해 직접 얻어 가는 것이 더 나은 방식으로 생각합니다.

``` c
#ifdef HAVE_ECORE_WAYLAND
  if (tzsh->toolkit_type == TZSH_TOOLKIT_TYPE_EFL)
   {
       // if client backend is efl, than get surface_id from window(Ecore_Wl2_Window id)
       Ecore_Wl2_Window *eww;
       eww = ecore_wl2_display_window_find(ecore_wl2_connected_display_get(NULL), win);
       if (eww) win = ecore_wl2_window_surface_id_get(eww);
    }
```

Reply:
저희는 최대한 native API변화가 없도록 작성했는데, 이 c# 코드에서 window id를 구하는게 이상하다고 판단하신다면, 
native구현에서 `Ecore_Wl2_Window`를 통해 직접 window id를 얻어가시는게 좋을 듯합니다.

Reply:
아 Ecore_Wl2_Window객체가 인자로 전달된다고 생각했었는데, 그것은 아니군요, 그렇다면 Window Id대신에 Ecore_Wl2_Window객체 자체를 전달하는 것은 어떤가요?

Reply:
여튼 IWindow자체도 특정 Window System에 의존성이 없도록 하고 싶은 구현인데, 그런 관점에서 Window ID가 window에 접근할 수 있는 유연한 값인가 Ecore_Wl2_Window객체가 더 유연한 값인가 비교하였고, 타 F/W의 API에서 사용되고 있는 타입을 참고하여 Ecore_Wl2_Window핸들을 전달하기로 결정하였습니다. 

여기에 Window ID를 추가하는 것은 중복된 정보를 제공하면서도 다른 윈도우 시스템이 되었을 때, Window ID가 의미 없는 값이 될수 있어서 적합하지 않다고 판단합니다.

Reply:
wiertel님이 말한 코멘트는 TizenShell에서 IWindowProvider를 제공하는게 어떻냐는 의미로 이해했습니다.
이에 TizenShell은 window정보를 가지지 않기에, TizenShell에서 window를 제공하는건 부정적라는 의미로 답을 드렸습니다.

@myroot 님께서 말씀하고자 하는 것은, Native API인 SoftkeyClient에서 Window ID 말고 Ecore_Wl2_Window를 인자로 받도록 변경하기를 원하시는건가요?

Reply:
@juns-kim
아 코멘트의 의도를 잘못 해석했었습니다. 
wiertel님의 코멘트와 별개로 이 변경사항에 대해서 언급한걸로 이해하여 잘못된 코멘트를 달았던 것 같습니다. 

말씀해주신데로, 제가 말하고자 한 것은 Window를 식별하기 위한 데이터가 가능한 하나로 통일되었으면 하는 의견으로 코멘트를 드렸습니다. 그러나 그렇게 하기에는 Native API의 변경사항 또한 발생하여 C# API구현 내에서 변환하는것에 대한 의견이었는데, 지금 보니 전혀 상관 없는 부분에 대한 코멘트여서 혼란이 된것 같아 죄송합니다.

Reply:
@myroot 넵 확인 감사드립니다 😄

Reply:
> Seems like it increases the coupling between TizenShell and Window. The TizenShell simply handles the Tizen-ws-shell protocol and doesn't contain any window information.

@wiertel 
Hello, could you check Mr. juns-kim's answer? 
I think it would be good to leave current interface.
thank you.

Reply:
If the TizenShell is not involved with window information then let's keep the current interface as it is.